### DISCUSSION TITLE
Hightouch can't sync with Supabase

### DISCUSSION THREAD

**Discussion by realestateintegrate** [supabase/supabase#30040] [Questions]
I started with the free plan on Supabase. I set everything up in Hightouch to connect and it can connect and see the database showing sample content from the tables. However, when I try to sync with the database to listen for new rows added or updated, the syncs continue to fail saying either the Connection Terminate Unexpectedly or json<>json. I tried upgrading to the pro account on Supabase and saw that now a few times it completed a sync successfully but now it continues to fail repeatedly. Supabase support says it's not looking like a resource issue since I'm under all the limits in my activity. I'm not getting much support from Supabase or Hightouch to know what the issue is or how to proceed. My entire workflow depends on the database connections and automations and I'm offline now for 2 days which is bad. Any help would be appreciated.

**Comment by GaryAustin1** [comment]
If you are using a supavisor connection, you might try the direct database port (ipv6 by default) but with an ipv4 option on Pro.  No evidence, just a another variable if you are using it.

**Comment by realestateintegrate** [comment]
We're connecting with the host, port, database name, user, etc. We can connect and see the database, just can't successfully sync. If we remove rows down to 200 it syncs which means to me that everything is set up properly, but when we try to do the full table which is only 321 rows, it fails. Even on the pro plan with the free upgrade from Nano to Micro, this shoudln't be a problem.

**Reply by GaryAustin1** [reply]
That did not address my response.  You can use the direct database port or you can use the Supavisor session port.  Both SHOULD act the same, but it is an extra layer of code on the connections. 
This shows the format of the direct connection versus the supavisor connection.
![image](https://github.com/user-attachments/assets/030d7b58-0626-4796-bfdc-69a35106bc85)

And I won't be of much help as I don't use direct connections in my code, and have not run across a user issue as a moderator on this.

**Reply by GaryAustin1** [reply]
You might also try using PSQL or a database IDE tool to download the table in full and see if it has an issue.

**Reply by realestateintegrate** [reply]
We were conencting to the port 6543 but when I try port 5432 the same issue persists.

**Reply by GaryAustin1** [reply]
With Supavisor or without?  There are two different 5432 ports.

**Reply by realestateintegrate** [reply]
![image](https://github.com/user-attachments/assets/84bf955e-ed7a-4cc6-a749-e9617c2aed30)

**Reply by realestateintegrate** [reply]
![image](https://github.com/user-attachments/assets/073d515e-244c-4def-a2ec-6108e402dac1)

**Reply by GaryAustin1** [reply]
That is WITH Supavisor.   Suggesting you test without to rule that out.

**Reply by realestateintegrate** [reply]
I'll try

**Reply by GaryAustin1** [reply]
Uncheck the box.  Note that is ipv6 only unless you have ipv4 option in Pro.  That may or may not impact you depending on Hightouch.

But just to be clear, I'm just throwing out a variable involved with connections and have no evidence this is involved.

**Reply by realestateintegrate** [reply]
There's no difference whether it's checked or not.

**Reply by GaryAustin1** [reply]
It is not a matter of checking it, but using the different URL.
It is just an interface to get you the different connection strings for supavisor 2 modes and the direct connect without supavisor.

**Reply by realestateintegrate** [reply]
No go with different credentials.
![GYvTbQPg5T](https://github.com/user-attachments/assets/8e419bd1-b4de-46e9-bb12-92dec42de2b7)

**Reply by GaryAustin1** [reply]
Yeah that could be the ipv4 versus ipv6 issue I mentioned, or some other string value.

As I was just trying to eliminate the Supavisor from being the issue, with no evidence it is, I'll just leave it to you.

There are no reports of issues with Hightouch and Supabase that I can find.     The dataports should have no issues though downloading a large table, and hundreds of rows is not much to have issues from the database or connections.

**Reply by GaryAustin1** [reply]
If you do not see any errors in the Supabase dashboard postgres logs about a timeout then it is not the DB.  But, for a select of hundreds of rows you are talking milliseconds, not seconds.

**Reply by GaryAustin1** [reply]
* assuming Hightouch is doing a single select to get the data versus 1 row at a time.   I have no idea what it does.

**Reply by GaryAustin1** [reply]
Also Supavisor does have a limit of 200 connections at once.  If for some reason Hightouch is creating a new connection for each row to do what it is doing that could generate an issue.  But if it fails exactly at 200 sort of implies it would be leaving each open.   That would not be a great way for it to connect to a database though, versus 1 connection and do the things it needs to do.

**Reply by realestateintegrate** [reply]
So now, without me or my developer changing anything, and with Hightouch confirming they've done nothing. Everything is now working as it should with all the same settings unchanged from what I set up initially. Just ran it again and now it's all fine. How this happens after 2 days of trying, with no one telling me they changed anything, is actually concerning that it means it could happen again at any time since the issue was not actually fixed.

**Comment by realestateintegrate** [comment]
My developer and I don’t know what more to do. We believe everything is set up properly. Tried with Lightning and Basic in Hightouch. Works fine with small amount of rows but even 321 is pretty small. Also worked fine when our database was hosted elsewhere. Something seems to timeout after querying the rows since we can see it counts 321 rows but continues spinning then change from “Completed” to “Failed” on Hightouch. Regards,Andrew St. Hilairewww.realestateintegrate.com(204) 955-8496On Oct 22, 2024, at 3:59 PM, GaryAustin1 ***@***.***> wrote:﻿
Yeah that could be the ipv4 versus ipv6 issue I mentioned, or some other string value.
As I was just trying to eliminate the Supavisor from being the issue, with no evidence it is, I'll just leave it to you.
There are no reports of issues with Hightouch and Supabase that I can find.     The dataports should have no issues though downloading a large table, and hundreds of rows is not much to have issues from the database or connections.
